### DISCUSSION TITLE
package-install help

### DISCUSSION THREAD

**Discussion by irorochad** [fuma-nama/fumadocs#879] [Q&A]
hey, can you please tell me how to set up 
```
package-install
```

?

I want to be able to do something like this..
<img width="861" alt="Screenshot 2024-09-19 at 3 57 44 PM" src="https://github.com/user-attachments/assets/d2888b28-f054-4d8a-b0f6-7781dbd6d133">

**Comment by irorochad** [comment]
I found it [https://fumadocs.vercel.app/docs/headless/mdx/install](https://fumadocs.vercel.app/docs/headless/mdx/install)

UPDATE: It didn't work... @fuma-nama 

My next config file

```
import { createMDX } from "fumadocs-mdx/next";
import { remarkInstall } from "fumadocs-docgen";

const withMDX = createMDX();

/** @type {import('next').NextConfig} */
const config = {
  reactStrictMode: true,
  remarkPlugins: [remarkInstall],
};

export default withMDX(config);

```

`/docs/page.tsx`

```
// other codes
  return (
    <DocsPage
      toc={page.data.toc}
      full={page.data.full}
      tableOfContent={{
        style: "clerk",
        single: false,
      }}
    >
      <DocsTitle>{page.data.title}</DocsTitle>
      <DocsDescription>{page.data.description}</DocsDescription>
      <DocsBody>
        <MDX components={{ ...defaultMdxComponents, Tab, Tabs }} />
      </DocsBody>
    </DocsPage>
  );
  // other codes
```

Error: image attached:

<img width="1440" alt="Screenshot 2024-09-19 at 4 53 48 PM" src="https://github.com/user-attachments/assets/a0cd946a-7016-4332-8b6b-de380c0bceab">

**Reply by fuma-nama** [reply]
On Fumadocs MDX v10, you should pass remark plugins to source config

**Reply by irorochad** [reply]
thanks. Fixed it!

**Reply by rodrigomf24** [reply]
@fuma-nama  @irorochad  where is the source config?

**Reply by fuma-nama** [reply]
See https://fumadocs.vercel.app/docs/mdx

**Reply by rodrigomf24** [reply]
Thanks @fuma-nama!  As feedback I would suggest updating [this](https://fumadocs.vercel.app/docs/headless/mdx/install) documentation (which seems to be outdated) to have a reference to the source.config.ts or a callout indicating the config should be added in the source.config.ts file, i.e:

```
// source.config.ts

import { defineDocs, defineConfig } from 'fumadocs-mdx/config';
import { remarkInstall } from 'fumadocs-docgen';

export const { docs, meta } = defineDocs({
  dir: 'content/docs',
});

export default defineConfig({
  mdxOptions: {
    remarkPlugins: [remarkInstall],
  }
});
```

cheers!

**Reply by fuma-nama** [reply]
That plugin is universal and can be used without Fumadocs MDX, for example: next-mdx-remote and some other libraries powered by MDX compiler, I would love to ensure it's simple enough for other sources

**Reply by fuma-nama** [reply]
Think I'll improve it like:

https://github.com/user-attachments/assets/983297a8-fc24-4f4c-948e-f07f22a85f6f

Welcome for other feedback too

**Reply by rodrigomf24** [reply]
@fuma-nama  ship it! Also building docs with fuma-docs is a breeze, thus far I've been loving it.
